plugin/rewrite/rewrite.go

```diff
@@ -100,12 +97,12 @@ func newRule(args ...string) (Rule, error) {
 		startArg = 1
 	}
 
-	if ruleType != "edns0" && expectNumArgs != 3 {
+	if ruleType != "edns0" && ruleType != "name" && expectNumArgs != 3 {
 		return nil, fmt.Errorf("%s rules must have exactly two arguments", ruleType)
 	}
 	switch ruleType {
 	case "name":
-		return newNameRule(args[startArg], args[startArg+1])
+		return newNameRule(mode, args[startArg:]...)
 	case "class":
 		return newClassRule(args[startArg], args[startArg+1])
```

Comment:
Unrelated to this PR, but just seeing that these don't use `mode`. Does the continue/stop thing not work for these rule types? The docs imply otherwise.

Reply:
Will check.

Reply:
Prior to this PR the name rule always returned “stop” when match was found. Here, i added NextAction to the rules. Then, on Mode(), the function returns the value in the rule.

Now, stop/continue actually works. 

Will rebase and push in the evening.

Reply:
Yes. Can you do the same for the other rule types in a separate PR?

Reply:
>  Yes. Can you do the same for the other rule types in a separate PR?

@johnbelamaric , sure. Please create an issue to I can create a PR against it.

Reply:
#1307

Reply:
Perfect. Will create a PR after the merge of this one.